workflows/prognostic_run_diags/fv3net/artifacts/query.py

```diff
@@ -0,0 +1,142 @@
+import argparse
+import asyncio
+import dataclasses
+import datetime
+import os
+import pathlib
+
+import fsspec
+
+import gcsfs
+
+
+@dataclasses.dataclass
+class Step:
+    fs: fsspec.AbstractFileSystem
+    path: pathlib.Path
+    date_created: datetime.date
+    project: str
+    tag: str
+    step: str
+
+
+async def flat_gather(promises):
+    out = await asyncio.gather(*promises)
+    return sum(out, [])
```

Comment:
Nit: This is clever though it also took me a few to realize what it was doing. Is there a way it could be made more obvious that `out` is a list of lists?

Reply:
I added type hints and used a more explicit name for `out`.